apps/src/applab/RecordListener.js

```diff
@@ -0,0 +1,230 @@
+'use strict';
+
+/**
+ * Class which allows callers to listen to changes to data tables. This is currently
+ * done by polling. Some modest rate-limiting is achieved by waiting for one second
+ * after the previous response is processed for data from a given table, rather than
+ * polling on a fixed interval and allowing a backlog to accumulate.
+ *
+ * The most logical incremental performance improvement is to use pusher to only
+ * re-read from the table if any of its contents have changed. Down the road, we
+ * may be able to use DynamoDB Streams / Lambda / SQS to deliver fine-grained
+ * notifications to the client without ever needing to  re-read the entire table.
+ *
+ * @constructor
+ */
+var RecordListener = module.exports = function () {
+  /**
+   * Map from table name to table handler.
+   * @private {Object.<string, TableHandler>}
+   */
+  this.tableHandlers_ = {};
+};
```

Comment:
I think we want a closure surrounding the rest of the code in this module so that we don't end up polluting the global namespace (e.g. with RECORD_INTERVAL)